pkg/operator/controller/ingress/dns.go

```diff
@@ -110,6 +107,14 @@ func desiredWildcardDNSRecord(ic *operatorv1.IngressController, service *corev1.
 		target = ingress.IP
 	}
 
+	dnsPolicy := iov1.ManagedDNS
+
+	// Set the DNS management policy on the dnsrecord to "Unmanaged" if ingresscontroller has "Unmanaged" DNS policy or
+	// if the ingresscontroller domain isn't a subdomain of the cluster's base domain.
+	if ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy == operatorv1.UnmanagedLoadBalancerDNS || !domainMatchesBaseDomain {
```

Comment:
Is it necessary to check `domainMatchesBaseDomain` here if `Reconcile` or `setDefaultPublishingStrategy` sets `ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy`?  I'm assuming `Reconcile` will correctly set `ic.Status.EndpointPublishingStrategy.LoadBalancer.DNSManagementPolicy` if an older version of the operator left it unset.